gradio-app/gradio
# Show view api button even if there are no named api routes in the app

# Description

Fixes #2160

### To test

- [x] Case of no named api routes: run `demo/calculator_blocks/run.py`

Should see the "To add api docs" page

<img width="1379" alt="image" src="https://user-images.githubusercontent.com/41651716/190218393-b75b3e94-1532-4b1b-9829-da8a656cddd5.png">

- [x] Case of named api routes: run modify `demo/calculator_blocks/run.py`   and add an api_name to `submit_btn.click` 

Should see the actual api docs

<img width="1415" alt="image" src="https://user-images.githubusercontent.com/41651716/190218564-42c96ed1-9829-457f-93c4-aa62400b1b19.png">

- [x] Case of suppressing the api page when there are named api routes: Set `show_api` to False in the above modified `calculator_blocks` demo

<img width="1404" alt="image" src="https://user-images.githubusercontent.com/41651716/190219920-b6be1608-85c7-4759-be4e-5a641baeda9f.png">

You can also do this with an interface demo like `calculator`


# Checklist:
